clippy_dev/src/lib.rs

```diff
@@ -56,22 +57,22 @@ impl Lint {
     }
 }
 
-pub fn collect_all() -> Vec<Lint> {
+pub fn gather_all() -> impl Iterator<Item=Lint> {
```

Comment:
this entire function can probably be `lint_files().flat_map(identity)`

Reply:
This is my favorite refactoring in Ruby and I'm SO happy it's possible in Rust :party: